internal/provider/resource_directory_role.go

```diff
@@ -49,10 +51,13 @@ __Further documentation:__
 					uuidvalidator.ValidUUID(),
 				},
 			},
-			"id": schema.StringAttribute{ // required hashicorps terraform plugin testing framework
-				DeprecationMessage:  "Use the `directory_id` attribute instead",
-				MarkdownDescription: "The ID of the directory.",
+			"id": schema.StringAttribute{ // required by hashicorps terraform plugin testing framework
+				DeprecationMessage:  "Use the `directory_id`, `name`, `app_id` and `role_template_name` attributes instead",
```

Comment:
The order as concatenated in the "id" is `directory_id`, `name`,  `role_template_name` and `app_id`. Don't know if this should be the same.

Reply:
Good catch, fixed.

Reply:
probably you want to do the same for subaccount? sorry for only commenting for directory.

Reply:
Now we have it fixed